### ISSUE TITLE
playlist error empty playlist name

### ISSUE THREAD

**Issue by seb2010** [iobroker-community-adapters/ioBroker.spotify-premium#94]
the adapter throws a "playlist error empty playlist name" from time to time:
`spotify-premium.0 2021-01-18 15:32:01.073 error (9995) playlist error empty playlist name`

I looked into the code and the default value in main.js in line 829 for the plaliste name is '', therefore empty and the error is thrown. The if statement or the default value of '' has to be changed.
`let fn = function(item) {`
`    	let playlistName = loadOrDefault(item, 'name', '');`
`       if (isEmpty(playlistName)) {`

**Comment by gcaruso73** [comment]
Same problem.
Do you already have a solution?

**Comment by m0uh85** [comment]
Same here :(

**Comment by obakuhl** [comment]
I had the same problem. I think this was due to an API change from Spotify. Since yesterday evening (18.00) the error no longer occurs and all playlists are visible in ioBroker.

**Comment by m0uh85** [comment]
> I had the same problem. I think this was due to an API change from Spotify. Since yesterday evening (18.00) the error no longer occurs and all playlists are visible in ioBroker.

I have still the problem

**Comment by TiDO2k** [comment]
i also have the problem.

**Comment by GermanBluefox** [comment]
Please test 1.1.4

**Comment by TiDO2k** [comment]
problem still there
playlist error empty playlist name

**Comment by TiDO2k** [comment]
`
spotify-premium.0 | 2021-11-17 17:32:23.929 | error | playlist error empty playlist name
-- | -- | -- | --
spotify-premium.0 | 2021-11-17 17:32:23.926 | warn | empty playlist name

`

problem still there

**Comment by seb2010** [comment]
same here

**Comment by GermanBluefox** [comment]
Ich kann da nicht reproduzieren..... Und deswegen nicht fixen

**Comment by m0uh85** [comment]
> Ich kann da nicht reproduzieren..... Und deswegen nicht fixen

Also ich habe vor längerem einfach alle Playlisten gelöscht und auch auf meinen Devices mal Ordnung gemacht und seit dem keine Probleme mehr.

**Comment by TiDO2k** [comment]
```

spotify-premium.0 | 2021-11-18 09:17:13.263 | error | playlist error empty playlist name
-- | -- | -- | --
host.iobrok | 2021-11-18 09:16:27.834 | error | instance system.adapter.spotify-premium.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
spotify-premium.0 | 2021-11-18 09:16:27.161 | error | Cannot read property 'val' of null
spotify-premium.0 | 2021-11-18 09:16:27.161 | error | TypeError:  Cannot read property 'val' of null    at fn  (/opt/iobroker/node_modules/iobroker.spotify-premium/main.js:1194:31)     at Array.map (<anonymous>)    at refreshPlaylistList  (/opt/iobroker/node_modules/iobroker.spotify-premium/main.js:1199:29)     at /opt/iobroker/node_modules/iobroker.spotify-premium/main.js:764:21
spotify-premium.0 | 2021-11-18 09:16:27.161 | error | unhandled promise rejection: Cannot read property 'val' of null
spotify-premium.0 | 2021-11-18 09:16:27.160 | error | Unhandled  promise rejection. This error originated either by throwing inside of  an async function without a catch block, or by rejecting a promise which  was not handled with .catch().
spotify-premium.0 | 2021-11-18 09:16:27.159 | error | playlist error empty playlist name
spotify-premium.0 | 2021-11-18 09:01:27.264 | error | playlist error empty playlist name
```
hier nochmal ein neues LOG. hilft dir das weiter?

**Comment by GermanBluefox** [comment]
> ```
> 
> spotify-premium.0 | 2021-11-18 09:17:13.263 | error | playlist error empty playlist name
> -- | -- | -- | --
> host.iobrok | 2021-11-18 09:16:27.834 | error | instance system.adapter.spotify-premium.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
> spotify-premium.0 | 2021-11-18 09:16:27.161 | error | Cannot read property 'val' of null
> spotify-premium.0 | 2021-11-18 09:16:27.161 | error | TypeError:  Cannot read property 'val' of null    at fn  (/opt/iobroker/node_modules/iobroker.spotify-premium/main.js:1194:31)     at Array.map (<anonymous>)    at refreshPlaylistList  (/opt/iobroker/node_modules/iobroker.spotify-premium/main.js:1199:29)     at /opt/iobroker/node_modules/iobroker.spotify-premium/main.js:764:21
> spotify-premium.0 | 2021-11-18 09:16:27.161 | error | unhandled promise rejection: Cannot read property 'val' of null
> spotify-premium.0 | 2021-11-18 09:16:27.160 | error | Unhandled  promise rejection. This error originated either by throwing inside of  an async function without a catch block, or by rejecting a promise which  was not handled with .catch().
> spotify-premium.0 | 2021-11-18 09:16:27.159 | error | playlist error empty playlist name
> spotify-premium.0 | 2021-11-18 09:01:27.264 | error | playlist error empty playlist name
> ```
> 
> hier nochmal ein neues LOG. hilft dir das weiter?

Das sind die Logs von NICHT aktueller Version. Kannst du die aktuelle ausprobieren? 1.1.7

**Comment by seb2010** [comment]
Du meinst Version 1.1.4? Folgende logs:
<html>
<body>


spotify-premium.0 | 2021-11-18 18:12:00.149 | error | Cannot read property 'length' of undefined
-- | -- | -- | --
spotify-premium.0 | 2021-11-18 18:12:00.149 | error | TypeError: Cannot read property 'length' of undefined at fn (/opt/iobroker/node_modules/iobroker.spotify-premium/main.js:773:39) at Array.map (<anonymous>) at deleteUsersPlaylist (/opt/iobroker/node_modules/iobroker.spotify-premium/main.js:795:29) at /opt/iobroker/node_modules/iobroker.spotify-premium/main.js:761:24
spotify-premium.0 | 2021-11-18 18:12:00.148 | error | unhandled promise rejection: Cannot read property 'length' of undefined
spotify-premium.0 | 2021-11-18 18:12:00.147 | error | Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
spotify-premium.0 | 2021-11-18 18:12:00.124 | error | playlist error empty playlist name
spotify-premium.0 | 2021-11-18 18:12:00.115 | warn | empty playlist name
spotify-premium.0 | 2021-11-18 17:57:01.351 | error | playlist error empty playlist name
spotify-premium.0 | 2021-11-18 17:57:01.345 | warn | empty playlist name
spotify-premium.0 | 2021-11-18 17:55:48.086 | info | starting. Version 1.1.4 in /opt/iobroker/node_modules/iobroker.spotify-premium, node: v12.22.1, js-controller: 3.3.18


</body>
</html>

**Comment by GermanBluefox** [comment]
Es gibt 1.1.8

**Comment by seb2010** [comment]
ich steh dann auf dem Schlauch wie ich diese Version installiert bekomme....

**Comment by m0uh85** [comment]
> ich steh dann auf dem Schlauch wie ich diese Version installiert bekomme....

iobroker - Expertenmodus einschalten (So Männchen mit Glatze ganz oben) - Installieren aus eigener URL (dieses Katzen Symbol) - Benutzerdefiniert (dort die URL: "https://github.com/iobroker-community-adapters/ioBroker.spotify-premium" eintragen und installieren)

**Comment by seb2010** [comment]
tritt auch in Version 1.1.8 auf:
<html>
<body>


spotify-premium.0 | 2021-11-19 12:01:09.389 | error | playlist error empty playlist name
-- | -- | -- | --
spotify-premium.0 | 2021-11-19 12:01:09.381 | warn | empty playlist name
spotify-premium.0 | 2021-11-19 11:46:04.278 | error | playlist error empty playlist name
spotify-premium.0 | 2021-11-19 11:46:04.277 | warn | empty playlist name
spotify-premium.0 | 2021-11-19 11:44:54.027 | info | starting. Version 1.1.8 in /opt/iobroker/node_modules/iobroker.spotify-premium, node: v12.22.1, js-controller: 3.3.18


</body>
</html>

**Comment by badsnoopy667** [comment]
I think I found the reason / solution:
The error is exactly what it say's: You have a Spotify Playlist that has no name. Go to your Spotify playlists and look closely. Likely you will find one that has no name (but only the desciption). Re-Name or delete this playlist and the adapter works again.
Had this issue also and found, that my wife managed to add a playlist without a name... don't ask me how.
Deleted the playlist and now the adapter works again and shows also all other playlists again!

Maybe the adapter could be changed so that it can handle empty playlist names?

**Comment by sysadmin0815** [comment]
> I think I found the reason / solution: The error is exactly what it say's: You have a Spotify Playlist that has no name. Go to your Spotify playlists and look closely. Likely you will find one that has no name (but only the desciption). Re-Name or delete this playlist and the adapter works again. Had this issue also and found, that my wife managed to add a playlist without a name... don't ask me how. Deleted the playlist and now the adapter works again and shows also all other playlists again!
> 
> Maybe the adapter could be changed so that it can handle empty playlist names?
> 
> However, bit of a different topic: I still loose connection to Spotify with 1.1.9. every hour. Any idea why this is?

You are right. I had a playlist without name added to my lists. removed it, restarted adapter and the error was gone.
thanks man!

**Comment by fiesch** [comment]
I would like to add that the same appearantly happens if you use folders in Spotify. I had this error but no empty playlist name. I moved all playlist out of my folders into the "root" and the error went away

**Comment by khnz** [comment]
have the same issue. I checked all my playlists in the Spotify client, simplified the names (e.g., removed special characters), deleted problematic playlists, etc., but without success. I then looked at the returned object and noticed that the list of playlists contains completely empty entries (null). A JSON.stringify(parseJson) at the beginning of the createPlaylists function, for example, returns:
```
{
	"href": "https://api.spotify.com/v1/users/USERID/playlists?offset=0&limit=30",
	"limit": 30,
	"next": "https://api.spotify.com/v1/users/USERID/playlists?offset=30&limit=30",
	"offset": 0,
	"previous": null,
	"total": 35,
	"items": [
		null,
		{
			...
			"id": "abc",
			"images": [
				...
			],
			"name": "My Playlist 1",
			"owner": {
				...
			},
			...
			"type": "playlist",
			...
		},
		{
			...
			"id": "zxy",
			"images": [
				...
			],
			"name": "My Playlist 2",
			"owner": {
				...
			},
			...
			"type": "playlist",
			...
		},
		null,
		null,
		null,
		{
			...
			"id": "zxy",
			"images": [
				...
			],
			"name": "My Playlist n",
			"owner": {
				...
			},
			...
			"type": "playlist",
			...
		},
		null,
		null,
		null,
		null,
		...
	]
}
```

I therefore changed line 1006 from p = p to if (parseJson.items[i]) p = p. After that, the error no longer occurred, and my playlists are now listed in the data points again.

Does anyone know where these "null" entries come from?

**Comment by aruttkamp** [comment]
@khnz - Please create a new issue - this one is closed since 2.5 year ;-) 
please provide us with version used, errortext and if possible log files
